File: ruby/codeeval/149_zero_string_to_int.rb
def ti(inp)
  f = nil
  r = ''
  inp.split(' ').each do |z|
    if !f
        f = z
    else
      if f == '00'
        r += '1' * z.length
      else
        r += z
      end
      f = nil
    end
  end
  puts r.to_i(2)
end

File.open(ARGV[0]).each_line do |line|
  ti(line)
end

File: python/codeeval/149_zero_string_to_int.py
import sys

def ti(inp):
    f = None
    r = ''
    for z in inp.split(' '):
        if f is None:
            f = z
        else:
            if f == '00':
                r += '1' * len(z)
            else:
                r += z
            f = None
    return int(r, 2)

with open(sys.argv[1], 'r') as test_cases:
    for test in test_cases:
        print(ti(test))
